unit/src/main/java/org/apache/falcon/unit/LocalExtensionManager.java

```diff
@@ -18,35 +18,69 @@
 
 package org.apache.falcon.unit;
 
+import org.apache.commons.io.IOUtils;
 import org.apache.falcon.FalconException;
 import org.apache.falcon.entity.v0.Entity;
 import org.apache.falcon.entity.v0.EntityType;
+import org.apache.falcon.extensions.store.ExtensionStore;
 import org.apache.falcon.resource.APIResult;
-import org.apache.falcon.resource.AbstractExtensionManager;
+import org.apache.falcon.resource.proxy.ExtensionManagerProxy;
 import org.apache.falcon.security.CurrentUser;
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 /**
  * A proxy implementation of the extension operations in local mode.
  */
-public class LocalExtensionManager extends AbstractExtensionManager {
+public class LocalExtensionManager extends ExtensionManagerProxy {
     public LocalExtensionManager() {}
 
     public APIResult submitExtensionJob(String extensionName, String jobName, InputStream config,
                                         Map<EntityType, List<Entity>> entityMap) throws FalconException, IOException {
-        submitEntities(extensionName, null, jobName, entityMap, config);
+
+        for(Map.Entry<EntityType, List<Entity>> entry : entityMap.entrySet()){
+            for(Entity entity : entry.getValue()){
+                submitInternal(entity, "falconUser");
```

Comment:
Feeds before processes, here too. May be create a method and call from both submit and submitAndSchedule.